src/sonic-yang-models/yang-models/sonic-system-port.yang

```diff
@@ -0,0 +1,87 @@
+module sonic-system-port {
+
+    yang-version 1.1;
+
+    namespace "http://github.com/sonic-net/sonic-system-port";
+    prefix system-port;
+
+    import sonic-port {
+        prefix port;
+    }
+
+    import sonic-types {
+        prefix stypes;
+    }
+
+    description "SYSTEM_PORT YANG Module for SONiC OS";
+
+    revision 2022-11-10 {
+        description "First Revision";
+    }
+
+    container sonic-system-port {
+
+        container SYSTEM_PORT {
+
+            description "SYSTEM_PORT table of config_db.json";
+
+            list SYSTEM_PORT_LIST {
+
+                key "hostname asic_name ifname";
+
+                leaf hostname {
+                    type stypes:hostname;
+                }
+
+                leaf asic_name {
+                    type string {
+                        pattern "[Aa]sic[0-4]";
+                    }
+                }
+
+                leaf ifname {
+                    type string;
+                }
+
+                leaf core_index {
+                    type uint8 {
+                        range 0..7;
+                    }
+                    description "Hardware core index a port is affiliated with.";
+                }
+
+                leaf core_port_index {
+                    type uint16;
+                    description "Local port index index to an ASIC core.";
+                }
+
+                leaf num_voq {
+                    type uint8 {
+                        range 1..8;
+                    }
+                    description "Number of VoQs associated with a port.";
+                }
+
+                leaf speed {
```

Comment:
Can you add description? And what's the unit? kbps?

Reply:
A VOQ is a Virtual Output Queue. Typically one VOQ is allocated per traffic class per port. The unit is the number of queues per port. For example, 8 corresponds to 8 VOQs per port. This number is not related to the speed or a port.

Reply:
Then why do we have speed for VOQ?

Reply:
This is a YANG model for a system port and not for a VOQ. A system port has many attributes, some of which are core_index (which core the port is associated with in a multicore ASIC), number of VOQs associated with the port, and the speed of the port.  

Are you asking about this description `Number of VoQs associated with a port`? Or are you asking about the description for `leaf speed`?

Reply:
I'm asking about leaf speed.

Reply:
Is this clear now?

Reply:
I can add a description for `leaf speed`. The units are are bps.

Reply:
Oops. I meant to say the units are in kbps. Max speed for future compatibility is 800Gbps.

Reply:
If unit is kbps, max speed is 800Mbps. Plese include unit in description.

Reply:
Sorry I'm not thinking straight. The listed range is 1..800000. So, units are million bits per second since 800,000 mbps == 800,000,000,000 bps.

Reply:
Ok, please add description with unit.